modules/shared/src/main/scala/org/tessellation/schema/metagraphMessage.scala

```diff
@@ -0,0 +1,17 @@
+package org.tessellation.schema
+
+import org.tessellation.schema.address.Address
+import org.tessellation.schema.transaction.TransactionOrdinal
+import org.tessellation.security.signature.Signed
+
+import derevo.cats.{eqv, show}
+import derevo.circe.magnolia.{decoder, encoder}
+import derevo.derive
+
+object metagraphMessage {
+  @derive(eqv, show, encoder, decoder)
+  case class AdminFeeKeyMessage(transactionOrdinal: TransactionOrdinal, adminFeeKeyAddress: Signed[Address])
```

Comment:
Please take a look how `TransactionReference` is constructed. IMO you need to replicate that, so:
```scala
case class MessageReference(ordinal: MessageOrdinal, hash: Hash)
```

Please don't reuse `TransactionOrdinal` since the message isn't the transaction itself.

Reply:
@ryle-goehausen, @marcinwadon Can you get together to discuss this? I'm not sure how this fits with what I was asked to do.

Reply:
If we want to follow the convention here established for wrapping it in a reference, that's fine, I don't think it will actually make a difference in this case though, since we're not using the previous message as a validation criteria for the new message, so it's irrelevant what the hash reference actually is (and whether or not the nodes agree on it.) The ordinal alone protects against replay attacks, so I don't think this serves a functional purpose other than organization.

Considering that this message acts as a transaction (maybe not our type, but the generic type,) and we want to introduce additional transaction types in the future, I don't think it makes sense to come up with a unique case class for each usage of TransactionOrdinal. If we want to include the hash though for bookkeeping purposes or other traversal or indexing purposes (or in case we want to change the behavior in the future to require some previous data validation), that might make sense though. So to me this seems more about organization, if it's easy to change then I'd rather just re-use transactionReference rather than creating a unique type for each future transaction we introduce.

Reply:
Ok. If so, I agree that the hash reference is not needed then. For clean organization, I'd definitely rename it to something else than `Transaction`. It's something you can send, but it's more an event rather than transaction. If you want to implement multiple type of transactions (we already have DAG and Currency transactions) it should be based on the same Transaction type.

Reply:
Created `MessageOrdinal` to address this